cadctl/cmd/investigate/investigate.go

```diff
@@ -40,19 +41,42 @@ var InvestigateCmd = &cobra.Command{
 	Use:          "investigate",
 	SilenceUsage: true,
 	Short:        "Filter for and investigate supported alerts",
-	RunE:         run,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+
+		if config.Config.ConfigFile != "" {
+			// loads the configuration unless config file is present
+			// config should be used only when developing
+			if _, err = config.Config.GetCadOcmClientID(cmd); err != nil {
+				return fmt.Errorf("failed to get cadOcmClientID: %w", err)
+			}
+			if _, err = config.Config.GetCadOcmClientSecret(cmd); err != nil {
+				return fmt.Errorf("failed to get cadOcmClientSecret: %w", err)
+			}
+			if _, err = config.Config.GetCadOcmURL(cmd); err != nil {
+				return fmt.Errorf("failed to get cadOcmURL: %w", err)
+			}
+		}
+
+		if _, err = config.Config.GetExperimental(cmd); err != nil {
+			return fmt.Errorf("failed to get experimental flag: %w", err)
+		}
+
+		return nil
+	},
+	RunE: run,
 }
 
 var (
-	logLevelFlag = ""
-	payloadPath  = "./payload.json"
+	payloadPath = "./payload.json"
 )
 
 const pagerdutyTitlePrefix = "[CAD Investigated]"
 
 func init() {
+	logging.RawLogger = logging.InitLogger(config.Config.LogLevel, "")
```

Comment:
We're initializing the `logging.RawLogger` variable twice now: once in `logging.go` and once here.

Reply:
We're now initializing it twice before initializing it with the cluster_id, so a third time

Reply:
I have a followup on this that focuses entirely on logging.